UPGRADE-GUIDE.md

```diff
@@ -178,7 +178,13 @@ dropped entirely.
 Previously, `rack.input` was required to be rewindable, i.e. `io.seek(0)` but
 this was only generally possible with a file based backing, which prevented
 efficient streaming of request bodies. Now, `rack.input` is not required to be
-rewindable.
+rewindable. Use `Rack::RewindableInput::Middleware` to make the body rewindable.
+
+### `rack.input` is no longer rewound after being consumed
+
+Previously, `rack.input` was always rewindable, and `.rewind` was called after
+consuming form or multipart data. Call `.rewind` explicitly to consume input
```

Comment:
Is this true? 2.x code called rewind always, no? or is this gated by some other `if`?

https://github.com/rack/rack/blob/2-2-stable/lib/rack/request.rb#L456

Reply:
Is this true? 2.x code called rewind always, no? or is this gated by some other `if`?

https://github.com/rack/rack/blob/2-2-stable/lib/rack/request.rb#L456

Reply:
The "wrong" part is the part that tells the user to call `rewind`, as the request body in Rack 3 is not required to respond to `rewind`.  That's why I recommended having a separate subheading for `Rack::RewindableInput::Middleware`, because if you are using that, then the request body will respond to `rewind`.

Reply:
Updated, WDYT? Happy to wordsmith futher.